backend/src/apiserver/server/pipeline_server.go

```diff
@@ -573,6 +573,94 @@ func NewPipelineServer(resourceManager *resource.ResourceManager, options *Pipel
 	return &PipelineServer{resourceManager: resourceManager, httpClient: http.DefaultClient, options: options}
 }
 
+// Creates a pipeline and a pipeline version in a single transaction.
+// Supports v2beta1 behavior.
+func (s *PipelineServer) CreatePipelineAndVersion(ctx context.Context, request *apiv2beta1.CreatePipelineAndVersionRequest) (*apiv2beta1.Pipeline, error) {
+	if s.options.CollectMetrics {
+		createPipelineRequests.Inc()
+		createPipelineVersionRequests.Inc()
+	}
+
+	// URL is a required input for CreatePipelineAndVersion call
+	pipelineUrlStr := request.GetPipelineVersion().GetPackageUrl().GetPipelineUrl()
+	if pipelineUrlStr == "" {
+		return nil, util.NewInvalidInputError("Failed to create a pipeline due to missing pipeline URL")
+	}
+
+	// Get pipeline description
+	description := request.GetPipeline().GetDescription()
+	if description == "" {
+		description = request.GetPipelineVersion().GetDescription()
+	}
+
+	// Convert the input request. Fail fast if either pipeline or pipeline version is corrupted.
+	pipeline, err := toModelPipeline(request.GetPipeline())
+	if err != nil {
+		return nil, util.Wrap(err, "Failed to create a pipeline due to pipeline conversion error")
+	}
+	pipeline.Description = description
```

Comment:
The `description` may come from pipeline version, if the description field of pipeline is empty. Should we change the pipeline description in this case, since this will change the data user enter?

Reply:
This is a good point. Currently, UI does not have a way to set `DisplayName` and `Description` for a pipeline version. The fall-back logic is fairly confusing, so let me replace it with overwriting both fields in `PipelineVersion` from `Pipeline`. I will add a comment in the proto to be clear.